File: Java/labs/lab19_Observer/task_2_1/src/WeatherDataCenter.java
public class WeatherDataCenter {

    final private String location;

    private float temperature; // температура повітря
    private float humidity;    // вологість повітря
    private float pressure;    // атмосферний тиск

    public WeatherDataCenter(String location) {
        this.location = location;
    }

    public void setMeasurements(float temperature, float humidity, float pressure) {
        this.temperature = temperature;
        this.humidity = humidity;
        this.pressure = pressure;
    }

    public float getTemperature() {
        return temperature;
    }

    public float getHumidity() {
        return humidity;
    }

    public float getPressure() {
        return pressure;
    }
}

File: Cpp/lab19_Observer/task_2_1/src/weather_data_center.cpp
#include <iostream>

#include "weather_data_center.hpp"


std::string path = "resources/input01.txt";

WeatherDataCenter::WeatherDataCenter()
: weatherReader(new WeatherReader(path))
{}

WeatherDataCenter::~WeatherDataCenter()
{
  delete weatherReader;
}

bool WeatherDataCenter::update()
{
  WeatherEvent* weatherEvent = this->weatherReader->getEvent();

  if (weatherEvent == nullptr)
  {
    return false;
  }

  if (weatherEvent->getLocation() == "end")
  {
    delete weatherEvent;
    return true;
  }

  std::cout << "Update data: city: " << weatherEvent->getLocation()
            << ", temperature = " << weatherEvent->getTemperature()
            << ", humidity = " << weatherEvent->getHumidity()
            << ", pressure = " << weatherEvent->getPressure()
            << std::endl;

  delete weatherEvent;
  return false;
}
